site/_posts/2020-08-11-advanced-registration-form.md

```diff
@@ -0,0 +1,176 @@
+
+{% include _image.liquid src="/assets/img/blogs/advanced-registration-forms/form-second-step.png" alt="Adding fields to our second step." class="img-fluid" figure=false %}
+
+Feel free to rearrange the form fields within each step by clicking the arrows to move a field up or down. The form configuration specifies steps and field display order within those steps. If you need to move a field between steps, delete it from one step and add it to another. Each field can live in zero or one steps.
+
+To change field validation or other attributes, return to the "Fields" section to make your changes. When you're done tweaking the form to your liking, save it.
+
+## Specify the form for your application
+
+Now that you've created a form with custom fields, the next step is to specify which applications should use it. Forms and form fields can be reused in any application and any tenant. Since FusionAuth supports multiple tenants per instance, you can easily reuse the fields and forms you've created. For each application, simply assign the registration form to the application. Luckily, this is the next step for this tutorial.
+
+Navigate to the "Applications" tab and create a new FusionAuth application. As a reminder, an application is anything a user can sign into: a web application, native app, API, anything. In this case, you are building out the real estate application; pick something descriptive. How about "Real estate search"? 
+
+You must configure a redirect URL; this is where the user is sent when registration succeeds. To set this up, navigate to the "OAuth" tab of your application and enter "https://fusionauth.io" or another public website. For a production application, of course, you'd instead specify where a user should be sent after registration. Each application may have multiple redirect URLs. Configure where to send a user after successful registration by adding a `redirect_uri` to the registration form link. For this tutorial, you'll only add one and that will be automatically added to the registration URL when you are ready to test drive this from the user's perspective.
+
+On top of setting up the form, you must configure an application to allow users to register themselves. Otherwise, no users will be allowed to create their own accounts. Navigate to the "Registration" tab and enable "Self service registration". Then check the "Advanced" option and select the form you created. Before you click the "Save" button, you'll see a screen like this: 
+
+{% include _image.liquid src="/assets/img/blogs/advanced-registration-forms/application-with-custom-form-configured.png" alt="Specifying a custom registration form for our application." class="img-fluid" figure=false %}
+
+Return to the list of applications. Your form is configured, so the next step is to sign up as a user. 
+
+## Sign up
+
+The moment you've been working towards is here! In just a few clicks, you'll see the glory of your custom registration form.
```

Comment:
We could mention how they can view the form step(s) using the theme preview for styling or what not - prior to actually going through the registration flow for real. 

`Customizations > Themes > Preview` - select the registration template, and this will allow you to select an application and step to view how it will render.

Reply:
OMG, that's awesome. I didn't know about that.

Reply:
I think it makes more sense to put that in the next blog post, which makes some theme changes.